### PR TITLE
[5.1] Added path option to migration create command

### PR DESCRIPTION
**Pull request by rubemlrm**
Added option to migration create command so users can specify a custom
directory to store migrations.

### CODE REVIEW

File: src/Illuminate/Database/Console/Migrations/BaseCommand.php [hunk 1]
```diff
@@ -9,9 +9,9 @@ class BaseCommand extends Command {
 	 *
 	 * @return string
 	 */
-	protected function getMigrationPath()
+	protected function getMigrationPath($path = NULL)
```
Comment (overtrue) [inline src/Illuminate/Database/Console/Migrations/BaseCommand.php]: `NULL` should be `null`

**Comment by Rubemlrm** [comment]
fixing that and other stuff i've missed like variable declartion on method

File: src/Illuminate/Database/Console/Migrations/BaseCommand.php [hunk 2]
```diff
@@ -9,9 +9,9 @@ class BaseCommand extends Command {
 	 *
 	 * @return string
 	 */
-	protected function getMigrationPath()
+	protected function getMigrationPath($path = null)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/BaseCommand.php]: You need to update the phpdoc.

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php [hunk 3]
```diff
@@ -84,7 +86,7 @@ public function fire()
 	 * @param  bool    $create
 	 * @return string
 	 */
-	protected function writeMigration($name, $table, $create)
+	protected function writeMigration($name, $table, $create, $path)
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php]: SAme here

**Comment by GrahamCampbell** [comment]
Please squash to one commit.

**Comment by Rubemlrm** [comment]
thanks for the tips i will do that =)

File: src/Illuminate/Database/Console/Migrations/BaseCommand.php [hunk 4]
```diff
@@ -6,12 +6,12 @@ class BaseCommand extends Command {
 
 	/**
 	 * Get the path to the migration directory.
-	 *
+	 * @param  string $path
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/BaseCommand.php]: string|null

Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/BaseCommand.php]: Also, you need **2** spaces before the dollar

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php [hunk 5]
```diff
@@ -116,6 +119,8 @@ protected function getOptions()
 			array('create', null, InputOption::VALUE_OPTIONAL, 'The table to be created.'),
 
 			array('table', null, InputOption::VALUE_OPTIONAL, 'The table to migrate.'),
+
+			array('path', null, InputOption::VALUE_OPTIONAL, 'The path where migration it will be stored.'),
```
Comment (Propaganistas) [inline src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php]: Fix language string to "The path where the migration will be stored."

**Comment by Rubemlrm** [comment]
@Propaganistas  thanks for the tip =) it's fixed

File: src/Illuminate/Database/Console/Migrations/BaseCommand.php (hunk 4 above)
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/BaseCommand.php]: Ping

File: src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php [hunk 6]
```diff
@@ -82,9 +84,10 @@ public function fire()
 	 * @param  string  $name
 	 * @param  string  $table
 	 * @param  bool    $create
+	 * @param  string|null  $path
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Console/Migrations/MigrateMakeCommand.php]: just string here, not string|null

**Comment by GrahamCampbell** [comment]
This doesn't even work, you're setting the path on top after passing it into the function?

**Pull request closed without merging**